api/v1/views/machine.py

```diff
@@ -42,15 +44,8 @@ def provider_filtered_machines(request, provider_uuid,
     Return all filtered machines. Uses the most common,
     default filtering method.
     """
-    try:
-        esh_driver = prepare_driver(request, provider_uuid, identity_uuid)
-    except Exception:
-        # TODO: Observe the change of 'Fail loudly' here
-        # and clean up the noise, rather than hide it.
-        logger.exception(
-            "Driver could not be prepared - Provider: %s , Identity: %s"
-            % (provider_uuid, identity_uuid))
-        esh_driver = None
+    # NOTE: raises many types of exceptions
+    esh_driver = prepare_driver(request, provider_uuid, identity_uuid)
```

Comment:
Are we losing valuable logging information by pulling `prepare_driver( ... )` out of the `try: .. except`? 

I'd think we've want to be logging the _types_ of exceptions so that we can go back to understand what causes this to misbehave. 

But we can come back to this.